app/actions/handlers.py

```diff
@@ -411,6 +425,23 @@ async def action_reprocess_file(integration, action_config: ReprocessFileConfig)
     logger.info(f"Executing reprocess_file action with integration {integration} and action_config {action_config}...")
     integration_id = str(integration.id)
     file_name = action_config.filename
+
+    # check current file status
+    file_status = await get_file_status(file_name)
+
+    if not file_status:
+        msg = f"File '{file_name}' not found. Skipping reprocessing."
+        logger.warning(msg)
+        return {"observations_processed": 0, "message": msg}
+    if file_status.get("status") == FileStatus.IN_PROGRESS.value:
+        msg = f"File '{file_name}' processing is already in progress. Skipping reprocessing."
+        logger.warning(msg)
+        return {"observations_processed": 0, "message": msg}
+    if file_status.get("status") == FileStatus.PROCESSED.value:
+        msg = f"File '{file_name}' was already processed and deleted. Please contact Gundi team to restore it."
+        logger.warning(msg)
+        return {"observations_processed": 0, "message": msg}
+
```

Comment:
I think this one is ok, so someone can restore it from the gcp console and then re-run the action

Reply:
But then it may be a good idea to modify `action_set_file_status(..)` for the case when no status is found. If the status is lost we can call `state_manager.group_add(..) ` directly to force-set the status to pending and then re-process the file.